src/main/java/com/onarandombox/MultiverseSignPortals/utils/PortalDetector.java

```diff
@@ -227,38 +226,49 @@ public Entity[] getRedstoneTeleportEntities(Sign sign) {
-                    xOff = -radius;
-                } else if (data[2].equals("south") || data[2].equals("SOUTH")) {
-                    xOff = radius;
-                } else if (data[2].equals("east") || data[2].equals("EAST")) {
-                    zOff = -radius;
-                } else if (data[2].equals("west") || data[2].equals("WEST")) {
-                    zOff = radius;
-                } else if (data[2].equals("up") || data[2].equals("UP")) {
-                    yOff = radius;
-                } else if (data[2].equals("down") || data[2].equals("DOWN")) {
-                    yOff = -radius;
+                switch (data[2].toUpperCase()) {
+                    case "NORTH":
+                        xOff = -radius;
+                        break;
+                    case "SOUTH":
+                        xOff = radius;
+                        break;
+                    case "EAST":
+                        zOff = -radius;
+                        break;
+                    case "WEST":
+                        zOff = radius;
+                        break;
+                    case "UP":
+                        yOff = radius;
+                        break;
+                    case "DOWN":
+                        yOff = -radius;
+                        break;
                 }
             }
             Vector signVector = sign.getBlock().getLocation().toVector();
             Vector min = new Vector(signVector.getX() + (xOff - radius), signVector.getY() + (yOff - radius), signVector.getZ() + (zOff - radius));
             Vector max = new Vector(signVector.getX() + (xOff + radius), signVector.getY() + (yOff + radius), signVector.getZ() + (zOff + radius));
             List<LivingEntity> worldEntities = sign.getBlock().getWorld().getLivingEntities();
             List<LivingEntity> entitiesInRange = new ArrayList<LivingEntity>(worldEntities.size());
+
             for (LivingEntity entity : worldEntities) {
-                if ((type == ALL || type == ANIMALS) && (entity instanceof Animals || entity instanceof Squid || entity instanceof Villager)) {
-                    if (entity.getLocation().toVector().isInAABB(min, max)) {
-                        Logging.finest("Found " + entity + " within range!");
-                        entitiesInRange.add(entity);
-                    }
-                } else if ((type == ALL || type == MONSTERS) && (entity instanceof Monster)) {
-                    if (entity.getLocation().toVector().isInAABB(min, max)) {
-                        Logging.finest("Found " + entity + " within range!");
-                        entitiesInRange.add(entity);
-                    }
-                } else if ((type == ALL || type == PLAYERS) && (entity instanceof HumanEntity)) {
+                boolean tryToAddEntity = false;
+
+                if (type == RedstoneTeleportType.ALL) {
+                    tryToAddEntity = true;
```

Comment:
Previously the ALL check was also checking if they were Animals, Squids, Villagers, Monsters, or HumanEntities. Are there no LivingEntities we would not want to include?

Reply:
I guess display entities and armour stands?

Reply:
Checking for living entity sounds like a good shout

Reply:
We already check if the entity is a `LivingEntity` earlier on line 258 by casting, so there is no need to check again